### PR TITLE
✨ new(checkbox): checkbox nested state changes.

### PR DESCRIPTION
**Pull request by emgk**
Continuing #156 here

### CODE REVIEW

**Comment by abder** [comment]
@emgk this is ready

File: packages/ui/checkbox/src/checkbox.types.ts [hunk 1]
```diff
@@ -1,60 +1,100 @@
...
-	 * A callback function to handle checkbox button selection changes.
+	 *
+	 * A callback function to handle checkbox button selection changes, with two signatures
+	 * - A Single argument which represent a list of items
+	 * - 3 Arguments for a single item: id, isChecked and groupId
+	 *
 	 */
-	onChange?(value: CheckboxValueType): void
+	onChange?(id: string, isChecked: boolean, groupId: boolean | string): void
+	onChange?(items: CheckBoxItemsList): void
```
Comment (emgk) [inline packages/ui/checkbox/src/checkbox.types.ts:58]: @abder duplicate method keys, can we fix please?

File: packages/ui/checkbox/src/checkbox.tsx [hunk 2]
```diff
@@ -1,137 +1,110 @@
...
+		uuid = id
+	}
 
-		// use ID from props list if exists
-		if (!!id) {
-			uuid = id
-		}
+	useEffect(() => {
+		// Add the checkbox details to the context list on component mount
+		ctx?.addToList(id as string, isChecked, groupId)
```
Comment (emgk) [inline packages/ui/checkbox/src/checkbox.tsx:38]: @abder can we fix this error ? 
image

File: packages/ui/checkbox/src/checkbox-context.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+	// Define a callback function triggered by the Checkboxes onChange event
+	const onChangeCallback = (
+		id: CheckboxItemTypes["id"],
+		isChecked?: CheckboxItemTypes["isChecked"],
+		groupId?: CheckboxItemTypes["groupId"],
+	) => {
+		// Trigger the onChange method from CheckboxContext, if it exists
+		ctx?.onChange?.(id, isChecked, groupId)
```
Comment (emgk) [inline packages/ui/checkbox/src/checkbox-context.tsx:92]: image

**Review by emgk — COMMENTED** [review]
@abder beside comments, there are some type warnings, can we get them check please? other than those it looks good to me.

**Pull request merged**